src/App.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import classNames from 'classnames';
+import { GoodsList } from './GoodsList';
+import './App.css';
+
+const goodsFromServer = [
+  'Dumplings',
+  'Carrot',
+  'Eggs',
+  'Ice cream',
+  'Apple',
+  'Bread',
+  'Fish',
+  'Honey',
+  'Jam',
+  'Garlic',
+];
+
+class App extends React.Component {
+  state = {
+    goods: [],
+  };
+
+  showGoods = () => {
+    this.setState({
+      goods: [...goodsFromServer],
+    });
+  }
```

Comment:
Let's change logic here a bit. It would be better to create in the state the value, for example, `areGoodsVisible`, and render 'show' button and `GoodsList` according to this value:

`{this.state.areGoodsVisible && <GoodsList />}`
and
`{!this.state.areGoodsVisible && <button>}`
When button is clicked you should change that value:

```
showGoods = () => {
  this.setState(state => {(
      areGoodsVisible: !state.areGoodsVisible
    )}
  )
}

```